api/v1_claim_rewards.go

```diff
@@ -225,65 +206,169 @@ func getValidatorAttestation(args GetValidatorAttestationParams) (*SenderAttesta
 }
 
 // Gets reward claim attestations from AAO and Validators in parallel.
+// Now handles validator selection internally with retry logic for failures.
 func fetchAttestations(
 	ctx context.Context,
 	rewardClaim RewardClaim,
-	validators []string,
+	allValidators []config.Node,
+	excludedOperators []string,
 	antiAbuseOracle config.Node,
 	signature string,
 	hasAntiAbuseOracleAttestation bool,
+	minVotes int,
 ) ([]SenderAttestation, error) {
 
+	// Shuffle the validators for random selection
+	shuffled := slices.Clone(allValidators)
+	rand.Shuffle(len(shuffled), func(i, j int) {
+		shuffled[i], shuffled[j] = shuffled[j], shuffled[i]
+	})
+
+	// Track which operators we've already used (including excluded ones)
+	usedOperators := make(map[string]bool)
+	for _, excluded := range excludedOperators {
+		usedOperators[excluded] = true
+	}
+
+	// Track which specific validators we've marked as bad
+	badValidators := make(map[string]bool)
+
 	offset := 0
 	if !hasAntiAbuseOracleAttestation {
 		offset = 1
 	}
-	attestations := make([]SenderAttestation, len(validators)+offset)
 
-	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	// We need minVotes - len(excludedOperators) successful validator attestations
+	validatorsNeeded := minVotes - len(excludedOperators)
+	attestations := make([]SenderAttestation, 0, validatorsNeeded+offset)
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
-	innerGroup, _ := errgroup.WithContext(ctx)
+
+	// Start AAO attestation in parallel if needed
+	var aaoAttestation *SenderAttestation
+	var aaoErr error
+	aaoDone := make(chan struct{})
 
 	if !hasAntiAbuseOracleAttestation {
-		innerGroup.Go(func() error {
+		go func() {
```

Comment:
nit: what's your take on managing channels directly like this vs errgroup, and whether we should mix etc? I think it might make sense despite this being a single task to keep the errgroup to be consistent and worry less about channel management? wdyt?

Reply:
i dont have a strong pref. i went back & forth, but ultimately decided to just do a single channel for this because it's always just one oracle attestation.
fine w/ keeping as a group though, updated back